### PR TITLE
fix: no clarfication on team dashboard

### PR DESCRIPTION
**Pull request by cubercsl**

### CODE REVIEW

**Comment by vmcj** [comment]
@cubercsl for what type of clarifications was this a fix?

**Comment by cubercsl** [comment]
Clarification associates either problem or contest. The existing code selects `contestproblem` and `contest`, which selects nothing :(. So team will see nothing on their team home, but if they enable notification, they can click it and jump to the clarification URL and see its content.

**Comment by cubercsl** [comment]
This is our hotfix between warmup and contest, but this may not occur in demoweb, I will verify this later.

**Comment by vmcj** [comment]
> Clarification associates either problem or contest. The existing code selects `contestproblem` and `contest`, which selects nothing :(. So team will see nothing on their team home, but if they enable notification, they can click it and jump to the clarification URL and see its content.

From a quick glance at the code I would assume the `LeftJoin` would already have to do this. The only reason for not merging this would be the speed penalty though so let's see if we can reproduce this somewhere and I'll take a look later this week.

**Comment by cubercsl** [comment]
@vmcj I have repro this on demoweb, pls check it.

**Review by vmcj — APPROVED** [review]
I'll give the rest a bit of time to look at it but I agree its broken.

File: webapp/src/Controller/Team/MiscController.php [hunk 1]
```diff
@@ -103,8 +103,7 @@ public function homeAction(Request $request): Response
                 ->leftJoin('c.sender', 's')
                 ->leftJoin('c.recipient', 'r')
                 ->select('c', 'cp', 'p')
-                ->andWhere('c.contest = :contest')
-                ->andWhere('cp.contest = :contest')
+                ->andWhere('c.contest = :contest OR cp.contest = :contest')
```
Comment (eldering) [inline webapp/src/Controller/Team/MiscController.php:106]: General question about the existing code:
When would ever `cp.contest` match `$contest`, but `c.contest` not? That is, why can't we just have
```
->andWhere('c.contest = :contest')
```

Comment (nickygerritsen) [reply webapp/src/Controller/Team/MiscController.php:106]: When you unlink a problem from a contest is the only way I can think of.

**Comment by cubercsl** [comment]
@Dup4 told me that in his shadow (of my instance), clarification (to all) can be displayed correctly on the team homepage.
I suspect that in the current code, the clarification created by the shadow is inconsistent with the behavior created by the user from the UI.

File: webapp/src/Controller/Team/MiscController.php (hunk 1 above)
Comment (moesoha) [reply webapp/src/Controller/Team/MiscController.php:106]: this was implemented in 0219d8e173540e26b5867d219753dcb23f08fcc5

**Comment by moesoha** [comment]
I found that we misunderstood the motivation of the `->andWhere('cp.contest = :contest')` after `->andWhere('c.contest = :contest')`, so this PR should be altered.

At first, I think `->andWhere('cp.contest = :contest')` is used to support a problem in multiple contests (like one problem exists in Div.1 and Div.2). By this way, clarification can be decoupled with contests, and can be associated with either Contest or Problem. And a clarification associated with a problem can be synced to all contests.

But since I investigated the codes today, I found the commit implements above codes: 0219d8e173540e26b5867d219753dcb23f08fcc5. This code fixes the issue mentioned in #2279, but I don't think this is a solution. Obviously, @vmcj misunderstood the usage of LEFT JOIN, and the fix for this issue is actually an accident. I'd like to draft a new PR  to fix #2279 again.

**Pull request closed without merging**